### PR TITLE
CATROID-796: AdMob bricks

### PR DESCRIPTION
**Pull request by ndrnour**
**extra changes:**
* upgrade  google play service to 19.0.0 (except analytics)

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [ ] Include the name of the Jira ticket in the PR’s title
- [ ] Include a summary of the changes plus the relevant context
- [ ] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no compiler or linter warnings
- [ ] Perform a self-review of the changes
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Confirm that new and existing unit tests pass locally
- [ ] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s gitflow workflow
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by Walcho1125 — CHANGES_REQUESTED** [review]
Please look at the remarks

File: catroid/src/main/java/org/catrobat/catroid/content/admob/AdMobBanner.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    val adView: AdView = AdView(context)
+    // private val deviceId: String = Utils.getAdMobDeviceId(context)
+
+    fun createNew(adSize: AdSize?, relativeLayoutRule: Int) {
+        adView.adSize = adSize
+        adView.adUnitId = BuildConfig.ADMOB_BANNER_UNIT_ID
+        // val testDeviceIds = listOf(deviceId)
+        val configuration = RequestConfiguration.Builder()
+            .setTagForChildDirectedTreatment(TAG_FOR_CHILD_DIRECTED_TREATMENT_TRUE)
+            // .setTestDeviceIds(testDeviceIds)
```
Comment (Walcho1125) [inline catroid/src/main/java/org/catrobat/catroid/content/admob/AdMobBanner.kt:50]: are the comments intended

File: Jenkinsfile [hunk 2]
```diff
@@ -66,6 +73,8 @@ pipeline {
         booleanParam name: 'BUILD_ALL_FLAVOURS', defaultValue: false, description: 'When selected all flavours are built and archived as artifacts that can be installed alongside other versions of the same APK.'
         string name: 'DEBUG_LABEL', defaultValue: '', description: 'For debugging when entered will be used as label to decide on which slaves the jobs will run.'
         string name: 'DOCKER_LABEL', defaultValue: '', description: 'When entered will be used as label for docker catrobat/catroid-android image to build'
+        string name: 'ADMOB_BANNER_APP_ID', defaultValue:'ca-app-pub-5619801003513533~9206664779', description: 'add some text'
+        string name: 'ADMOB_BANNER_UNIT_ID', defaultValue:'ca-app-pub-5619801003513533/1184704529', description: 'add some text'
```
Comment (Walcho1125) [inline Jenkinsfile:77]: please add a proper description

**Review by Walcho1125 — CHANGES_REQUESTED** [review]
please take a look at the remarks

File: catroid/src/main/java/org/catrobat/catroid/content/admob/AdMobBanner.kt (hunk 1 above)
Comment (Walcho1125) [reply catroid/src/main/java/org/catrobat/catroid/content/admob/AdMobBanner.kt:50]: Please also remove the other comments (https://confluence.catrob.at/display/KNOWHOW/Clean+code)

**Review by Walcho1125 — APPROVED** [review]
Well done!

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/AdMobBannerBricks.kt [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            sizeSpinner = this
+            setOnItemSelectedListener(this@AdMobShowBannerBrick)
+            setSelection(size)
+        }
+
+        return view
+    }
+
+    override fun addRequiredResources(requiredResourcesSet: Brick.ResourcesSet) {
+        requiredResourcesSet.add(Brick.SPEECH_RECOGNITION)
```
Comment (wslany) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/AdMobBannerBricks.kt:143]: Why?

File: catroid/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -863,6 +863,18 @@
...
+
+    <!-- AdMob Bricks Strings -->
+    <string name="brick_hide_admob_banner">Hide AdMob banner</string>
+    <string name="brick_show_admob_banner">Show AdMob banner at:</string>
+    <string name="brick_admob_show_banner_size">With size</string>
+    <string name="brick_admob_position_top">Top</string>
+    <string name="brick_admob_position_bottom">Bottom</string>
+    <string name="brick_admob_size_banner">BANNER</string>
+    <string name="brick_admob_size_smart_banner">SMART_BANNER</string>
+    <string name="brick_admob_size_large_banner">LARGE_BANNER</string>
```
Comment (wslany) [inline catroid/src/main/res/values/strings.xml:876]:
```suggestion
    <string name="brick_hide_admob_banner">Hide AdMob banner</string>
    <string name="brick_show_admob_banner">Show AdMob banner at</string>
    <string name="brick_admob_show_banner_size">with size</string>
    <string name="brick_admob_position_top">top</string>
    <string name="brick_admob_position_bottom">bottom</string>
    <string name="brick_admob_size_banner">standard</string>
    <string name="brick_admob_size_smart_banner">smart</string>
    <string name="brick_admob_size_large_banner">large</string>
```

File: Jenkinsfile [hunk 5]
```diff
@@ -43,6 +43,13 @@ def webTestUrlParameter() {
     return env.WEB_TEST_URL?.isEmpty() ? '' : "-PwebTestUrl='${params.WEB_TEST_URL}'"
 }
 
+def adMobParameters() {
+    return !env.ADMOB_BANNER_APP_ID?.isEmpty() && !env.ADMOB_BANNER_UNIT_ID?.isEmpty() ?
+            "-PadMobBannerAppId='${env.ADMOB_BANNER_APP_ID}' -PadMobBannerUnitId='${env.ADMOB_BANNER_UNIT_ID}'"
+            : "-PadMobBannerAppId=ca-app-pub-5619801003513533~9206664779 " +
+            "-PadMobBannerUnitId=ca-app-pub-5619801003513533/1184704529"
```
Comment (wslany) [inline Jenkinsfile:50]: How did we create these test ids?

File: Jenkinsfile [hunk 6]
```diff
@@ -66,6 +73,8 @@ pipeline {
         booleanParam name: 'BUILD_ALL_FLAVOURS', defaultValue: false, description: 'When selected all flavours are built and archived as artifacts that can be installed alongside other versions of the same APK.'
         string name: 'DEBUG_LABEL', defaultValue: '', description: 'For debugging when entered will be used as label to decide on which slaves the jobs will run.'
         string name: 'DOCKER_LABEL', defaultValue: '', description: 'When entered will be used as label for docker catrobat/catroid-android image to build'
+        string name: 'ADMOB_BANNER_APP_ID', defaultValue:'ca-app-pub-5619801003513533~9206664779', description: 'AdMob banner App id'
+        string name: 'ADMOB_BANNER_UNIT_ID', defaultValue:'ca-app-pub-5619801003513533/1184704529', description: 'AdMob banner unit id'
```
Comment (wslany) [inline Jenkinsfile:77]: Duplicate strings?

File: catroid/gradle/standalone_apk_tasks.gradle [hunk 7]
```diff
@@ -231,3 +231,18 @@ tasks.whenTaskAdded { task ->
     }
 }
 
+ext.getAdMobAppID = { ->
+    if (project.hasProperty('adMobBannerAppId')) {
+        return project['adMobBannerAppId']
+    } else {
+        return "ca-app-pub-5619801003513533~9206664779"
```
Comment (wslany) [inline catroid/gradle/standalone_apk_tasks.gradle:238]: Duplicate numbers see comments above?

File: catroid/gradle/standalone_apk_tasks.gradle [hunk 8]
```diff
@@ -231,3 +231,18 @@ tasks.whenTaskAdded { task ->
...
+    }
+}
+
+ext.getAdMobUnitID = { ->
+    if (project.hasProperty('adMobBannerUnitId')) {
+        return project['adMobBannerUnitId']
+    } else {
+        return "ca-app-pub-5619801003513533/1184704529"
```
Comment (wslany) [inline catroid/gradle/standalone_apk_tasks.gradle:246]: Again?

**Review by wslany — CHANGES_REQUESTED** [review]
AdMob bricks shall not have their own AdMob brick category (now: dark red, at the bottom of all categories), but instead the bricks shall be part of the green Looks category (at the very bottom of the category), and the bricks shall be in green. 

There shall be a setting in the options for an "AdMob extension" with a submenu, and in the submenu a checkbox and an "Help" link that points to https://catrob.at/AdMob (this page in the wiki still needs to be created, so at the moment this is just a short-url that will then be redirected --- please use this short-URL). The checkbox allows to activate the bricks, i.e., only then will they become visible. The default shall be that they are not visible, i.e., the extension is deactivated.

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/AdMobBannerBricks.kt (hunk 3 above)
Comment (ndrnour) [reply catroid/src/main/java/org/catrobat/catroid/content/bricks/AdMobBannerBricks.kt]: nice catch! fixed

File: catroid/src/main/res/values/strings.xml (hunk 4 above)
Comment (ndrnour) [reply catroid/src/main/res/values/strings.xml]: fixed

File: Jenkinsfile (hunk 5 above)
Comment (ndrnour) [reply Jenkinsfile:50]: they are build params, see [jenkins build](https://jenkins.catrob.at/job/Catroid/job/PR-3917/36/parameters/)

File: Jenkinsfile (hunk 6 above)
Comment (ndrnour) [reply Jenkinsfile:77]: nope, these are default values to be shown in the text fields

File: catroid/gradle/standalone_apk_tasks.gradle (hunk 7 above)
Comment (ndrnour) [reply catroid/gradle/standalone_apk_tasks.gradle:238]: nope, these are values for building with Gradle, which should be removed if the property isn't specified by the user.

**Comment by ndrnour** [comment]
> AdMob bricks shall not have their own AdMob brick category (now: dark red, at the bottom of all categories), but instead the bricks shall be part of the green Looks category (at the very bottom of the category), and the bricks shall be in green.
> 
> There shall be a setting in the options for an "AdMob extension" with a submenu, and in the submenu a checkbox and an "Help" link that points to https://catrob.at/AdMob (this page in the wiki still needs to be created, so at the moment this is just a short-url that will then be redirected --- please use this short-URL). The checkbox allows to activate the bricks, i.e., only then will they become visible. The default shall be that they are not visible, i.e., the extension is deactivated.

ok, I'll fix that

**Comment by juliajulie95** [comment]
Is this still being worked at?

**Comment by ndrnour** [comment]
> Is this still being worked at?

it's been in code review for more than 2 months. just rebased.

**Comment by reichli** [comment]
closed due to inactivity

**Pull request closed without merging**